draft-ietf-quic-transport.md

```diff
@@ -2618,6 +2633,39 @@ advertise a smaller maximum ID.  A sender may receive MAX_STREAM_ID frames out
 of order; a sender MUST therefore ignore any MAX_STREAM_ID that does not
 increase the maximum.
 
+
+## Related Streams
+
+Using the Related Stream ID in the STREAM frame ({{frame-stream}}) allows for
+the creation of a pair of streams.  One endpoint creates a stream without
+assigning a related stream and the peer opens another stream that includes the
+stream ID of the first in the Related Stream ID of the first stream.  This
+enables straightforward use of QUIC for request-response protocols or protocols
+that rely on bidirectional channels.
+
+This document doesn't proscribe the use of the Related Stream ID field, leaving
+decisions about the semantics of related streams to protocols that use QUIC.
+Protocols that use QUIC MUST either define the semantics of related streams or
+prohibit the use of the Related Stream ID field.  Protocols MAY define other
+uses for the Related Stream ID field, but are advised that complex relationships
+between streams could be poorly supported in generic QUIC implementations.  The
+INVALID_RELATED_STREAM error code can be used by protocols to signal receipt of
+invalid Related Stream ID field values, unless the protocol defines a more
+specific error code for the circumstances.
+
+The Related Stream ID for a stream only needs to be sent once for each (allowing
+for retransmissions due to loss).  Sending the field on all stream frames with
+an offset of 0 ensures that the value is received without duplicating the field
+unnecessarily.  If the value of the Related Stream ID changes for a stream, the
+endpoint that receives the conflicting value MUST treat this as a connection
+error of type INVALID_RELATED_STREAM.
```

Comment:
OTOH - I can see the benefit of including the ID on every fram if you have a many to one relation because the recevier don't need to store a lot of context. But then, as I said before, it becomes impractical to generate an error. So alternatively a) don't set the R bit at all, b) set it only on the first frame (if app procotol says so), or c) set on all frames (if app protocol says so). In all cases the use of frame ids must be consistent and receiver SHOULD generate an error if it is able to do so.